src/main/java/org/apache/flink/connector/clickhouse/internal/executor/ClickHouseUpsertExecutor.java

```diff
@@ -96,12 +116,14 @@ public void setRuntimeContext(RuntimeContext context) {}
     public void addToBatch(RowData record) throws SQLException {
         switch (record.getRowKind()) {
             case INSERT:
```

Comment:
There is no need to invoke exist before insert statement, cause performance loss.